components/Dialogs/edit_task_dialog.py

```diff
@@ -18,9 +18,10 @@
 from PySide6 import QtWidgets
 
 class EditTaskDialog(QtWidgets.QDialog):
-    def __init__(self, description="", due="", priority=""):
+    def __init__(self, deletion_function, description="", due="", priority=""):
```

Comment:
Let's name this the action we are taking, not "delete function"

```suggestion
    def __init__(self, delete_task, description="", due="", priority=""):
```